### PR TITLE
Feature/map #118

### LINKED ISSUE #118: Add a Map!
Add a full screen map in the game!

Here is the folder for the map images: https://drive.google.com/open?id=0B1hH2izVwX3iZ2tyX1FoTk9xLVU

### PR DESCRIPTION
**Pull request by Fayyne**
resolves  #118 as a basic/phase 1 implementation.

A basic map in our full screen HUD, as well as a fix I found in the inventory code when I was playing around and thinking. 

Also made a slight change to the inventory filter popup selection menu. Its height just, felt bad, so I made it a little bigger. Looks fine at 1080p 1440p and smaller resolutions.

Let me know if anything needs to be changed for your repo and code standards.

### CODE REVIEW

File: game/hud/src/components/GameMenu/index.tsx [hunk 1]
```diff
@@ -156,6 +156,7 @@ export class GameMenu extends React.Component<GameMenuProps, GameMenuState> {
   }
 
   private fireVisibilityEvent = () => {
+    console.log('fire visibility event');
```
Comment (codecorsair) [inline game/hud/src/components/GameMenu/index.tsx]: remove the log here

File: game/hud/src/components/HUD/index.tsx [hunk 2]
```diff
@@ -7,7 +7,6 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
 import { client, ql, events } from 'camelot-unchained';
-import { useConfig } from 'camelot-unchained/lib/graphql/react';
```
Comment (codecorsair) [inline game/hud/src/components/HUD/index.tsx]: Why is this removed?

File: game/hud/src/components/HUDFullScreen.tsx [hunk 3]
```diff
@@ -190,11 +204,15 @@ class HUDFullScreen extends React.Component<FullScreenNavProps, FullScreenNavSta
   }
 
   private renderCharacter = (prop: { active: boolean }) => {
-    return <Character visibleComponent={this.state.visibleComponent} />;
+    return <Character visibleComponent={this.state.visibleComponent}/>;
```
Comment (codecorsair) [inline game/hud/src/components/HUDFullScreen.tsx]: Are you manually removing this white-space or using a plugin? The white-space here should be here as it was.

File: game/hud/src/widgets/Map/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  containerClass?: string;
+  visibleComponent: string;
+}
+
+export interface MapContainerState {
+  visible: boolean;
+  mapLoaded: boolean;
+}
+
+const map = new Image();
```
Comment (codecorsair) [inline game/hud/src/widgets/Map/index.tsx]: this should be a member of the class to make it more reusable.  You can initialize it in the constructor.

File: game/hud/src/widgets/Map/index.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  constructor(props: MapContainerProps) {
+    super(props);
+    this.state = {
+      visible: false,
+      mapLoaded: false,
+    };
+  }
+
+  public componentWillMount() {
+    map.src = 'images/world-map.jpg';
```
Comment (codecorsair) [inline game/hud/src/widgets/Map/index.tsx]: all of this in componentWillMount can and should be done in the constructor instead.

File: game/hud/src/components/HUD/index.tsx (hunk 2 above)
Comment (Fayyne) [reply game/hud/src/components/HUD/index.tsx]: not sure there, looking at git history, it was added by you, then removed during a merge. added back.

Comment (Fayyne) [reply game/hud/src/components/HUD/index.tsx]: OH I know what removed it, since it was unused it was auto removed do to some optimize settings I have.

File: game/hud/src/components/HUDFullScreen.tsx (hunk 3 above)
Comment (Fayyne) [reply game/hud/src/components/HUDFullScreen.tsx]: this was my auto format settings, I have resolved this I believe.
